### PR TITLE
Amcl refactorization

### PR DESCRIPTION
**Pull request by mhpanah**
Addressing issue's #79 and #88.

Note: Since the dynamic reconfigure is not enabled, I manually changed the default values to match turtlebot launch file default values.

TODO: Create issue to make sure formatting is correct. In this PR, I've only fixed the format in one place.

### CODE REVIEW

File: src/localization/nav2_amcl/src/amcl_node.cpp [hunk 1]
```diff
@@ -642,7 +651,7 @@ boost::recursive_mutex::scoped_lock ml(configuration_mutex_);
   while(!map_received)
   {
     auto result_future = client->async_send_request(req);
-    if (rclcpp::spin_until_future_complete(std::shared_ptr<rclcpp::Node>(this), result_future, std::chrono::seconds(1)) !=
+    if (rclcpp::spin_until_future_complete(std::shared_ptr<rclcpp::Node>(this, [](rclcpp::Node*){}), result_future, std::chrono::seconds(1)) !=
```
Comment (mjeronimo) [inline src/localization/nav2_amcl/src/amcl_node.cpp]: How does this lambda function with with spin_until_future_complete? Wondering whether it affects CPU utilization.

File: src/libs/nav2_util/src/map/map_draw.c [hunk 2]
```diff
@@ -25,6 +25,7 @@
  * CVS: $Id: map_draw.c 7057 2008-10-02 00:44:06Z gbiggs $
 **************************************************************************/
 
+#pragma GCC diagnostic ignored "-Wpedantic"
```
Comment (mkhansen-intel) [inline src/libs/nav2_util/src/map/map_draw.c]: Can we end the pragma after the offending include statements? I read that there is a 'push' and 'pop' for pragmas supported in some compilers to allow for this.

Comment (crdelsey) [reply src/libs/nav2_util/src/map/map_draw.c]: [GCC diagnostics docs](https://gcc.gnu.org/onlinedocs/gcc/Diagnostic-Pragmas.html)
**Summary**
```c++
#pragma GCC diagnostic push
#pragma GCC diagnostic ignored "-Wpedantic"
<code>
#pragma GCC diagnostic pop
```

Comment (mkhansen-intel) [reply src/libs/nav2_util/src/map/map_draw.c]: Yep, that's what I was thinking. @mhpanah can you add this, but only around the offending include statements. That way we can still catch warnings in our own code.

Comment (mhpanah) [reply src/libs/nav2_util/src/map/map_draw.c]: @mkhansen-intel Yes we can, but in this particular case the pragma is in the c file and it won't offend other files.

Comment (mhpanah) [reply src/libs/nav2_util/src/map/map_draw.c]: @mkhansen-intel Added the pragma around tf2/util.h file.

**Pull request merged**